samples/wifi/provisioning/prj.conf

```diff
@@ -0,0 +1,116 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+CONFIG_WIFI=y
+CONFIG_WIFI_NRF700X=y
+
+# WPA supplicant
+CONFIG_WPA_SUPP=y
+
+# System settings
+CONFIG_NEWLIB_LIBC=y
+CONFIG_NEWLIB_LIBC_NANO=n
+
+# Networking
+CONFIG_NETWORKING=y
+CONFIG_NET_SOCKETS=y
+CONFIG_POSIX_API=y
+CONFIG_NET_LOG=y
+CONFIG_NET_IPV6=n
+CONFIG_NET_IPV4=y
+CONFIG_NET_UDP=y
+CONFIG_NET_TCP=y
+CONFIG_NET_DHCPV4=y
+CONFIG_DNS_RESOLVER=y
+
+CONFIG_NET_STATISTICS=y
+
+CONFIG_NET_PKT_RX_COUNT=8
+CONFIG_NET_PKT_TX_COUNT=8
+
+# Below section is the primary contributor to SRAM and is currently
+# tuned for performance, but this will be revisited in the future.
+CONFIG_NET_BUF_RX_COUNT=16
+CONFIG_NET_BUF_TX_COUNT=16
+CONFIG_NET_BUF_DATA_SIZE=128
+CONFIG_HEAP_MEM_POOL_SIZE=153600
+CONFIG_NET_TC_TX_COUNT=0
+
+CONFIG_NET_IF_UNICAST_IPV4_ADDR_COUNT=1
+CONFIG_NET_MAX_CONTEXTS=5
+CONFIG_NET_CONTEXT_SYNC_RECV=y
+
+CONFIG_INIT_STACKS=y
+
+CONFIG_NET_L2_ETHERNET=y
+
+CONFIG_NET_CONFIG_SETTINGS=y
+
+CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.165.100.150"
+CONFIG_NET_CONFIG_PEER_IPV4_ADDR="192.165.100.1"
+
+CONFIG_NET_SOCKETS_POLL_MAX=4
+
+# Memories
+CONFIG_MAIN_STACK_SIZE=4096
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
+CONFIG_NET_TX_STACK_SIZE=4096
+CONFIG_NET_RX_STACK_SIZE=4096
+
+# Debugging
+CONFIG_STACK_SENTINEL=y
+CONFIG_DEBUG_COREDUMP=y
+CONFIG_DEBUG_COREDUMP_BACKEND_LOGGING=y
+CONFIG_DEBUG_COREDUMP_MEMORY_DUMP_MIN=y
+
+# Kernel options
+CONFIG_ENTROPY_GENERATOR=y
+
+# Logging
+CONFIG_LOG=y
+
+# Security
+# CONFIG_CRYPTO_MBEDTLS_SHIM=y
+CONFIG_ENTROPY_GENERATOR=y
+CONFIG_MBEDTLS=y
+CONFIG_NORDIC_SECURITY_BACKEND=y
+CONFIG_HARDWARE_DEVICE_CS_GENERATOR=y
+# TODO: We need to get it working with a standard configuration file
+CONFIG_GENERATE_MBEDTLS_CFG_FILE=n
+CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG=n
+# Need this for CTR_DBRG and ENTROPY
+CONFIG_MBEDTLS_CFG_FILE="src/nrf-config.h"
+CONFIG_MBEDTLS_USER_CONFIG_FILE="src/nrf-config-user.h"
+# To fix Kconfig MD5 warning
+CONFIG_NET_TCP_ISN_RFC6528=n
+
+CONFIG_NET_L2_NRF_WIFI_MGMT=y
+
+CONFIG_NET_MGMT_EVENT_STACK_SIZE=2048
+
+CONFIG_BT=y
+CONFIG_BT_DEBUG_LOG=y
+CONFIG_BT_SMP=y
+CONFIG_BT_PERIPHERAL=y
+
+CONFIG_BT_BUF_ACL_RX_SIZE=151
+CONFIG_BT_L2CAP_TX_MTU=147
+CONFIG_BT_BUF_ACL_TX_SIZE=151
+
+CONFIG_BT_RX_STACK_SIZE=4096
+CONFIG_BT_BONDABLE=n
```

Comment:
Why you then use API for bond removing?

Reply:
It seems that disabling this config only affects the flag when exchanging pairing request/response. The device will still keep bond info anyway, so API is used to remove the bond in disconnected callback.